examples/Cargo.toml

```diff
@@ -3,6 +3,7 @@ name = "hal-examples"
 version = "0.1.0"
 publish = false
 workspace = ".."
+edition = "2018"
```

Comment:
let's leave the edition update to other PRs, the scope is already high, unless you require it for this to work

Reply:
The code needs nll.